test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -233,24 +251,20 @@ public void initClient() throws IOException {
             assert adminClient == null;
             assert clusterHosts == null;
             assert availableFeatures == null;
-            assert nodeVersions == null;
+            assert nodesVersions == null;
             assert testFeatureService == null;
             clusterHosts = parseClusterHosts(getTestRestCluster());
             logger.info("initializing REST clients against {}", clusterHosts);
             client = buildClient(restClientSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
             adminClient = buildClient(restAdminSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
 
             availableFeatures = EnumSet.of(ProductFeature.LEGACY_TEMPLATES);
-            nodeVersions = new TreeSet<>();
-            var semanticNodeVersions = new HashSet<Version>();
+            nodesVersions = new TreeSet<>();
```

Comment:
How about wrapping this into an unmodifiable set, especially if getting exposed now?